docs/appendices/release-notes/5.4.3.rst

```diff
@@ -46,4 +46,6 @@ See the :ref:`version_5.4.0` release notes for a full list of changes in the
 Fixes
 =====
 
-None
+- Fixed a regression introduced with CrateDB 5.0.0 which can cause that some
```

Comment:
The bug is not related to the table statistics/join-order but if a hash-join is nested on placed on the right right-hand-side.

Reply:
```
Fixed a regression introduced with CrateDB 5.0.0 which can cause an empty result
when a hash-join has multiple relations placed on the right-hand side.
```

Reply:
I can not validate the issue from a SQL perspective by forcing a hash-join nested on the right side.
Example:
```
create table doc.t1(a int, b int);
create table doc.t2(c int, d int);
create table doc.t3(e int, f int);

insert into doc.t1(a,b) values(1,2);
insert into doc.t2(c,d) values (1,3),(5,6);
insert into doc.t3(e,f) values (3,2);

refresh table t1, t2, t3;
```

Note: NO `ANALYZE` is triggered.

Forcing a hash-join nested on the right side using sub-selects:
```
cr> explain SELECT t13.e FROM t2 JOIN (SELECT * FROM t1 JOIN t3 ON t1.b = t3.f) t13 ON t13.a = t2.c WHERE t2.d = t13.e;                                                                                                                                  
+-------------------------------------------+
| EXPLAIN                                   |
+-------------------------------------------+
| Eval[e]                                   |
|   └ HashJoin[((a = c) AND (d = e))]       |
|     ├ Collect[doc.t2 | [c, d] | true]     |
|     └ Rename[a, b, e, f] AS t13           |
|       └ HashJoin[(b = f)]                 |
|         ├ Collect[doc.t1 | [a, b] | true] |
|         └ Collect[doc.t3 | [e, f] | true] |
+-------------------------------------------+
EXPLAIN 1 row in set (0.004 sec)

cr> SELECT t13.e FROM t2 JOIN (SELECT * FROM t1 JOIN t3 ON t1.b = t3.f) t13 ON t13.a = t2.c WHERE t2.d = t13.e;                                                                                                                                          
+---+
| e |
+---+
| 3 |
+---+
SELECT 1 row in set (0.006 sec)
```

I agree that theoretically the issue is not related to join re-ordering but to unordered relation names (which is not deterministic but seems to happen for sure during join re-ordering).

Reply:
I just flipped the join conditions.

Reply:
Additionally, I think the issue can theoretically also occur with hash-joins nested on the left side due to wrong relation name ordering.

Reply:
> Try:
> 
> ```
> cr> SELECT t13.e FROM t2 JOIN (SELECT * FROM t1 JOIN t3 ON t1.b = t3.f) t13 ON t2.d = t13.e AND t13.a = t2.c;
> +---+
> | e |
> +---+
> +---+
> ```

I cannot verify this locally, neither with 5.3.4 nor 5.4.0, I get valid results. What do I miss?

Reply:
Sorry, that works.

Reply:
Ignore the last comments.

Reply:
I am also ok with the original note.